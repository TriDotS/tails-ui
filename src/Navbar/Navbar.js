import React, { Children } from 'react'

import PropTypes from 'prop-types'

const Navbar = ({ children }) => (
  <nav className="bg-white px-8 pt-2 shadow-md w-full">
    <div className="-mb-px flex justify-center">
      {Children.map(children, child => {
        return React.cloneElement(child, {
          className:
            'no-underline text-teal-dark uppercase tracking-wide font-bold text-xs py-3 mr-8'
        })
      })}
    </div>
  </nav>
)

Navbar.propTypes = {
  children: PropTypes.node
}

export default Navbar
