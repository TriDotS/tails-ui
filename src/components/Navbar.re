let component = ReasonReact.statelessComponent("Navbar");

let make = (~color, ~brand, ~navRight, ~children) => {
  ...component,
  render: _self =>
    <div className={j|bg-$(color) text-white h-24 flex flex-col|j}>
      <div className="mt-4 pl-4 flex flex flex-row align-center w-full">
        <div className="text-3xl uppercase font-black">
          (ReasonReact.stringToElement(brand))
        </div>
        <div className="ml-auto pr-4">
          (ReasonReact.arrayToElement(navRight))
        </div>
      </div>
      <div
        className="mt-4 pl-4 overflow-x-scroll whitespace-no-wrap flex flex-row">
        (ReasonReact.arrayToElement(children))
      </div>
    </div>
};

let default =
  ReasonReact.wrapReasonForJs(~component, jsProps =>
    make(
      ~color=jsProps##color,
      ~brand=jsProps##brand,
      ~navRight=jsProps##navRight,
      ~children=jsProps##children
    )
  );
