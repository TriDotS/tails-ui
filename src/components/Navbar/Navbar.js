import React, { Children } from 'react'

import PropTypes from 'prop-types'

const Navbar = ({ children, brand, color, navRight }) => (
  <div className={`bg-${color} text-white h-24 flex flex-col`}>
    <div className="flex items-center justify-between">
      <div className="mt-4 ml-4 flex flex flex-col w-full sm:w-full md:w-1/2 lg:w-1/2 xl:w-1/2">
        <div className="text-3xl">{brand}</div>
        <div className="mt-4 overflow-x-scroll">
          {Children.map(children, child => {
            return React.cloneElement(child, {
              className: 'text-white p-1 text-lg'
            })
          })}
        </div>
      </div>
      <div className="mr-4">{navRight}</div>
    </div>
  </div>
)

Navbar.defaultProps = {
  color: 'blue'
}

Navbar.propTypes = {
  children: PropTypes.node,
  brand: PropTypes.string,
  color: PropTypes.string,
  navRight: PropTypes.node
}

export default Navbar
