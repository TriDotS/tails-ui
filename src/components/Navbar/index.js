import React, { Children } from 'react'

import PropTypes from 'prop-types'

const Navbar = ({ color, brand, navRight, children }) => (
  <div className={`bg-${color} text-white h-24 flex flex-col`}>
    <div className="mt-4 pl-4 flex flex flex-row align-center w-full">
      <div className="text-3xl">{brand}</div>
      <div className="ml-auto pr-4">{navRight}</div>
    </div>
    <div className="mt-4 pl-4 overflow-x-scroll">
      {Children.map(children, child => {
        return React.cloneElement(child, {
          className: 'text-white p-1 text-lg'
        })
      })}
    </div>
  </div>
)

Navbar.defaultProps = {
  color: 'blue'
}

Navbar.propTypes = {
  color: PropTypes.string,
  brand: PropTypes.string,
  navRight: PropTypes.node,
  children: PropTypes.node
}

export default Navbar
