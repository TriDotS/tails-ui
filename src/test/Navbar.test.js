import Navbar from '../components/Navbar'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(
    <Navbar
      brand="Tails-UI"
      extra={
        <div>
          <a href="#!">Account</a>
          <div>Some Text</div>
        </div>
      }
    >
      <a href="#!">Home</a>
      <a href="#!">Products</a>
      <a href="#!">Cart</a>
      <a href="#!">Shop</a>
    </Navbar>,
    div
  )
})

test('navbar snapshot', () => {
  const tree = renderer
    .create(
      <Navbar
        brand="Tails-UI"
        extra={
          <div>
            <a href="#!">Account</a>
            <div>Some Text</div>
          </div>
        }
      >
        <a href="#!">Home</a>
        <a href="#!">Products</a>
        <a href="#!">Cart</a>
        <a href="#!">Shop</a>
      </Navbar>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
