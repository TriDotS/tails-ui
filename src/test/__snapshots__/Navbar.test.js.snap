// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar snapshot 1`] = `
<div
  className="bg-blue text-white h-24 flex flex-col"
>
  <div
    className="flex items-center justify-between"
  >
    <div
      className="mt-4 ml-4 flex flex flex-col w-full sm:w-full md:w-1/2 lg:w-1/2 xl:w-1/2"
    >
      <div
        className="text-3xl"
      >
        Tails-UI
      </div>
      <div
        className="mt-4 overflow-x-scroll"
      >
        <a
          className="text-white p-1 text-lg"
          href="#!"
        >
          Home
        </a>
        <a
          className="text-white p-1 text-lg"
          href="#!"
        >
          Products
        </a>
        <a
          className="text-white p-1 text-lg"
          href="#!"
        >
          Cart
        </a>
        <a
          className="text-white p-1 text-lg"
          href="#!"
        >
          Shop
        </a>
      </div>
    </div>
    <div
      className="mr-4"
    />
  </div>
</div>
`;
